src/reactionsystem_structure.jl

```diff
@@ -0,0 +1,1438 @@
+    ReactionSystem(fulleqs, t, usv, psv; spatial_ivs, continuous_events, discrete_events, observed, kwargs...)
+end
+
+
+### Base Functions ###
+
+"""
+    ==(rn1::ReactionSystem, rn2::ReactionSystem)
+
+Tests whether the underlying species, parameters and reactions are the same in
+the two [`ReactionSystem`](@ref)s. Requires the systems to have the same names
+too.
+
+Notes:
+- *Does not* currently simplify rates, so a rate of `A^2+2*A+1` would be
+    considered different than `(A+1)^2`.
+- Does not include `defaults` in determining equality.
+"""
+function (==)(rn1::ReactionSystem, rn2::ReactionSystem)
+    isequivalent(rn1, rn2; ignorenames = false)
+end
+
+"""
+    isequivalent(rn1::ReactionSystem, rn2::ReactionSystem; ignorenames = true)
+
+Tests whether the underlying species, parameters and reactions are the same in
+the two [`ReactionSystem`](@ref)s. Ignores the names of the systems in testing
+equality.
+
+Notes:
+- *Does not* currently simplify rates, so a rate of `A^2+2*A+1` would be
+    considered different than `(A+1)^2`.
+- Does not include `defaults` in determining equality.
+"""
+function isequivalent(rn1::ReactionSystem, rn2::ReactionSystem; ignorenames = true)
+    if !ignorenames
+        (nameof(rn1) == nameof(rn2)) || return false
+    end
+
+    (get_combinatoric_ratelaws(rn1) == get_combinatoric_ratelaws(rn2)) || return false
+    isequal(get_iv(rn1), get_iv(rn2)) || return false
+    issetequal(get_sivs(rn1), get_sivs(rn2)) || return false
+    issetequal(get_unknowns(rn1), get_unknowns(rn2)) || return false
+    issetequal(get_ps(rn1), get_ps(rn2)) || return false
+    issetequal(MT.get_observed(rn1), MT.get_observed(rn2)) || return false
+    issetequal(get_eqs(rn1), get_eqs(rn2)) || return false
+
+    # subsystems
+    (length(get_systems(rn1)) == length(get_systems(rn2))) || return false
+    issetequal(get_systems(rn1), get_systems(rn2)) || return false
+
+    true
+end
+
+
+### ModelingToolkit-inherited Functions ###
+
+# Retrives events.
+MT.get_continuous_events(sys::ReactionSystem) = getfield(sys, :continuous_events)
+# `MT.get_discrete_events(sys::ReactionSystem) = getfield(sys, :get_discrete_events)` should be added here.
+
+# need a custom equations since ReactionSystem.eqs are a mix of Reactions and Equations
+function MT.equations(sys::ReactionSystem)
+    ivs = independent_variables(sys)
+    eqs = get_eqs(sys)
+    systems = get_systems(sys)
+    if !isempty(systems)
+        eqs = CatalystEqType[eqs;
+                             reduce(vcat, MT.namespace_equations.(systems, (ivs,));
+                                    init = Any[])]
+        return sort!(eqs; by = eqsortby)
+    end
+    return eqs
+end
+
+function MT.unknowns(sys::ReactionSystem)
+    sts = get_unknowns(sys)
+    systems = get_systems(sys)
+    if !isempty(systems)
+        sts = unique!([sts; reduce(vcat, namespace_variables.(systems))])
+        sort!(sts; by = !isspecies)
+        return sts
+    end
+    return sts
+end
```

Comment:
These should be after the `get_*` functions as they are higher level and build from using `get_*` functions.

Reply:
That is also the order I had before...

Reply:
Can do. However, in this case I thought it would make more sense to have MTK-specific getters before Catalyst-specific ones.

Reply:
The `get_*` functions are more low-level compared to something like `unknowns` which is doing more than a direct field access (it aggregates across systems). Functions like `unknowns` are higher level and implemented in terms of getters, so it makes sense to define getters first (by my normal convention I was using before).